src/main/java/com/codejoust/main/service/ProblemService.java

```diff
@@ -243,6 +253,25 @@ public ProblemDto deleteProblem(String problemId, String token) {
         }
 
         service.verifyTokenMatchesUid(token, problem.getOwner().getUid());
+
+        /**
+         * Before the problem can be deleted, set all foreign key references
+         * to the Problem within all ProblemContainers to null.
+         * See https://stackoverflow.com/a/10030873/7517518.
+         */
+        List<ProblemContainer> problemContainers = problemContainerRepository.findAllByProblem(problem);
+        for (ProblemContainer problemContainer : problemContainers) {
+            problemContainer.setProblem(null);
+            problemContainerRepository.save(problemContainer);
+        }
```

Comment:
You're right, this can take a very long time - I agree, some method to optimize this would be very useful. This is especially a problem because after clicking the "Delete Problem" button, the page does not redirect until the REST call returns. I think we could address this issue in a future PR, though I'll leave a few ideas here:
1. Add a `boolean deletionInProgress` field that allows us to show a notice that the problem is currently being deleted on the frontend.
2. Redirect immediately after the call is made rather than waiting for the return.
3. Have the function return success if it parses through everything, even if it has not finished deleting the problem - this may require having the for loop above continue in a separate thread or something, which I'm fairly certain is possible though I don't know how that would really work.
4. Somehow speed up the actual deletion process.
5. Remove the `problem` relation and replace it with a `problemId`, then simply replace those with `null` whenever a reference is attempted and not found.